signature/policy_config_test.go

```diff
@@ -207,21 +217,41 @@ func TestDefaultPolicyPath(t *testing.T) {
 			},
 			false,
 			nondefaultPath,
+			false,
+			"",
 		},
 		// No environment expansion happens in the overridden paths
-		{&types.SystemContext{SignaturePolicyPath: variableReference}, false, variableReference},
+		{&types.SystemContext{SignaturePolicyPath: variableReference}, false, variableReference, false, ""},
+
+		{nil, false, "", false, fmt.Sprintf("no policy.json file found at any of the following: %q, %q", userDefaultPolicyPath, tempsystemdefaultpath)},
 	} {
-		if c.userfilePresent {
-			err := os.MkdirAll(filepath.Dir(userDefaultPolicyPath), os.ModePerm)
-			require.NoError(t, err)
-			f, err := os.Create(userDefaultPolicyPath)
-			require.NoError(t, err)
-			f.Close()
+		paths := []struct {
+			condition bool
+			path      string
+		}{
+			{c.userfilePresent, userDefaultPolicyPath},
+			{c.createSystemTempFile, tempsystemdefaultpath},
+		}
+
+		for _, p := range paths {
+			if p.condition {
+				err := os.MkdirAll(filepath.Dir(p.path), os.ModePerm)
+				require.NoError(t, err)
+				f, err := os.Create(p.path)
+				require.NoError(t, err)
+				f.Close()
+			} else {
+				os.Remove(p.path)
+			}
+		}
+		path, err := defaultPolicyPathWithHomeDir(c.sys, tempHome, tempsystemdefaultpath)
+		if c.expectedError != "" && err != nil {
+			assert.Empty(t, path)
```

Comment:
(If `err` is set, we are making no promises on `path`, it must not be used; this line, to the extent it might imply such a promise, is not really desirable and I think it’s better to remove it.)

Reply:
Sure, I understood your point.